src/App.js

```diff
@@ -1,34 +1,154 @@
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+  const [hasAccount, setHasAccount] = useState(true);
+  const [loading, setLoading] = useState(false);
+
+  
+  
+  const clearInputs = () =>{
+    setEmail('');
+    setPassword('');
+    setConfirmPassword('')
+  }
+  const clearErrors = () =>{
+    setEmailError('');
+    setPasswordError('');
+    setConfirmPasswordError('')
+  }
+const handleLogin = () =>{
+  setLoading(true)
+  clearErrors();
+  fire
+  .auth()
+  .signInWithEmailAndPassword(email, password).then((user)=>{
+    setLoading(false)
+    setUser(user)
+    clearInputs()
+  })
+  .catch(err =>{
+    setLoading(false)
+    switch(err.code){
+      case "auth/invalid-email":
+      case "auth/user-disabled":
+      case "auth/user-not-found":
+      setEmailError(err.message);
+      break;
+      case "auth/wrong-password":
+        setPasswordError(err.message);
+        break;
+    }
+  });
+  
+
+}
+
+const validateConfirmPassword=(password,confirmPassword )=>{
+  if(password==confirmPassword){
+    return true
+  }
+  else{
+    setConfirmPasswordError("Password doesn't match")
+    return false
+  }
+}
+
+const handlesignUp = () =>{
+  setLoading(true)
+  if(!(validateConfirmPassword(password, confirmPassword))){
+    setLoading(false)
+    return
+  }
+  clearErrors();
+  fire
+  .auth()
+  .createUserWithEmailAndPassword(email, password).then((user)=>{
+    setLoading(false)
+    setUser(user)
+    clearInputs()
+  })
+  .catch(err =>{
+    setLoading(false)
+    switch(err.code){
+      case "auth/email-already-in-use":
+      case "auth/invalid-email":
+      setEmailError(err.message);
+      break;
+      case "auth/weak-password":
+        setPasswordError(err.message);
+        break;
+    }
+  });
+ 
+
+}
+
+const resetPassword=(email)=>{
+  setLoading(true)
+   return fire.auth().sendPasswordResetEmail(email).then(()=>{ 
+    clearErrors()
+    clearInputs()
+    setLoading(false)
+  })
+}
+const handleLogout = ()=>{
+  fire.auth().signOut();
+};
+const authListener = ()=>{
+  fire.auth().onAuthStateChanged((user) => {
+    if(user){
+      clearInputs(); 
+   setUser(user);
+    }
+    else{
+      setUser("");
+    }
+  });
+}
+useEffect(() => {
+  authListener();
+},[]);
   return (
- <BrowserRouter>
- <div className = "Container">
- {/* <img src={logo} className ="App-logo" alt ="logo" /> */}
-   <h1 className = "m-3 d-flex justify-content-center">OC BLACKBOARD</h1>
-   
-   <Navigation/>
-   <Switch>
-     <Route path='/' component={Home} exact />
-     <Route path='/about' component={About} />
-     <Route path='/profile' component={Profile} />
-     <Route path='/courses' component={Courses} />
-     <Route path ='/messages' component ={Messages}/>
-     <Route path='/service-tests' component={ServiceTestPage} />
-   </Switch>
- </div>
- </BrowserRouter>
+    <div className = "App">
+      {user ? (
+        <Dashboard handleLogout={handleLogout} />
```

Comment:
Seems odd that `user` isn't passed to the Dashboard component as well (like it is to the Login component).  Maybe it would be nice to pass it along and display a basic acknowledgement of who the user is, such as a "Hi, {user.name}" somewhere.